File: java/parser/src/main/java/gov/lbl/parser/parser/Enumeration_literalVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Comment;
import gov.lbl.parser.domain.Enumeration_literal;

public class Enumeration_literalVisitor  extends modelicaBaseVisitor<Enumeration_literal> {
    @Override
    public Enumeration_literal visitEnumeration_literal(modelicaParser.Enumeration_literalContext ctx) {
        String identifier = ctx.IDENT().getText();
        CommentVisitor commentVisitor = new CommentVisitor();
        Comment comment = ctx.comment().accept(commentVisitor);
        return new Enumeration_literal(identifier, comment);
    }
}

File: jsParser/parser/Enumeration_literalVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Enumeration_literal = require('../domain/Enumeration_literal')

const CommentVisitor = require('./CommentVisitor')

class Enumeration_literalVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitEnumeration_literal (ctx) {
    let identifier = ''
    let comment = null

    if (ctx.IDENT()) {
      identifier = ctx.IDENT().getText()
    }

    if (ctx.comment()) {
      const commentVisitor = new CommentVisitor.CommentVisitor()
      comment = commentVisitor.visitComment(ctx.comment())
    }

    return new Enumeration_literal.Enumeration_literal(identifier, comment)
  }
};

Enumeration_literalVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitEnumeration_literal = this.visitEnumeration_literal
exports.Enumeration_literalVisitor = Enumeration_literalVisitor
